scripts/generate-version-info.sh

```diff
@@ -16,14 +16,14 @@ OUTPUT_FILE="$1"
 sudo rpm --query --all --queryformat '\{"%{NAME}": "%{VERSION}-%{RELEASE}"\}\n' | jq --slurp --sort-keys 'add | {packages:(.)}' > "$OUTPUT_FILE"
 
 # binaries
-KUBELET_VERSION=$(kubelet --version | awk '{print $2}')
+KUBELET_VERSION=$(sudo kubelet --version | awk '{print $2}')
```

Comment:
Can you just call the entire script with `sudo` in the template? https://github.com/awslabs/amazon-eks-ami/blob/8bb63f545e74e00eb3a2b919f147ad5cfce51725/eks-worker-al2.json#L227

Reply:
@cartermckinnon Unfortunately, it will not work correctly if you execute the template with `sudo`. Why? Because, the `generate-version-info.sh` would generate the `version-info.json` file but it will be owned by root. The following step will fail to download the file because of it's permissions (600). Keep in mind the I use hardened base AMI.

Execution logs:
```
2024-02-14T11:07:02Z: ==> amazon-ebs: Provisioning with shell script: /codebuild/output/src3189660426/src/repo/scripts/generate-version-info.sh
2024-02-14T11:07:04Z: ==> amazon-ebs: Downloading /tmp/worker/version-info.json => RYR-AL2-EKS-1.23-GOLDEN-AMI-v20240214-version-info.json
2024-02-14T11:07:05Z: ==> amazon-ebs: Download failed: scp: /tmp/worker/version-info.json: Permission denied
==> amazon-ebs:
2024-02-14T11:07:05Z: ==> amazon-ebs: Provisioning step had errors: Running the cleanup provisioner, if present...
2024-02-14T11:07:05Z: ==> amazon-ebs: Terminating the source AWS instance...
```

Reply:
Ah, gotcha. I'm fine with adding `sudo` to the commands, just looks silly to `sudo` just to run `aws --version`...

Reply:
I agree, looks stilly but does the job, ha ha.